data-formats/src/types.rs

```diff
@@ -1175,21 +1176,38 @@ impl KeyExchange {
         salt.extend_from_slice(&context_rand);
         salt.extend_from_slice(&((cipher.required_keylen() * 8) as u16).to_be_bytes());
 
-        #[cfg(feature = "use_noninteroperable_kdf")]
-        log::warn!("Using non-interoperable key derivation");
+        if !crate::interoperable_kdf_available() {
+            log::warn!("Forcing use of non-interoperable key derivation");
+            use_noninteroperable_kdf = true;
+        }
 
-        let kdf_args = [
+        let interoperable_kdf_args = [
             &KdfArgument::KbMode(KdfKbMode::Counter),
             &KdfArgument::Mac(KdfMacType::Hmac(cipher.kdf_digest())),
             &KdfArgument::Salt(KEY_DERIVE_LABEL),
             &KdfArgument::KbInfo(&salt),
             &KdfArgument::Key(&shared_secret),
-            #[cfg(not(feature = "use_noninteroperable_kdf"))]
             &KdfArgument::UseL(false),
-            #[cfg(not(feature = "use_noninteroperable_kdf"))]
             &KdfArgument::R(8),
         ];
-        let key_out = perform_kdf(KdfType::KeyBased, &kdf_args, cipher.required_keylen())?;
+        let noninteroperable_kdf_args = [
+            &KdfArgument::KbMode(KdfKbMode::Counter),
+            &KdfArgument::Mac(KdfMacType::Hmac(cipher.kdf_digest())),
+            &KdfArgument::Salt(KEY_DERIVE_LABEL),
+            &KdfArgument::KbInfo(&salt),
+            &KdfArgument::Key(&shared_secret),
+        ];
+        let key_out = perform_kdf(
+            KdfType::KeyBased,
+            if use_noninteroperable_kdf {
```

Comment:
Omg this is super nice using statements in function arguments

Reply:
Yep, absolutely!